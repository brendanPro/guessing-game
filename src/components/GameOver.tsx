import type { Pokemon } from "@/types/pokemon";
import { Card, CardContent } from "./ui/card";

interface GameOverProps {
    pokemon: Pokemon | null;
}

export function GameOver({ pokemon }: GameOverProps) {
    return (
        <Card className="border-red-500 bg-red-50 dark:bg-red-950">
          <CardContent className="p-4 text-center">
            <div className="text-red-600 dark:text-red-400">
              <div className="text-2xl font-bold mb-2">ðŸ˜ž Game Over</div>
              <div>Le Pokemon Ã©tait <strong>{pokemon?.frenchName}</strong></div>
            </div>
          </CardContent>
        </Card>
    );
}