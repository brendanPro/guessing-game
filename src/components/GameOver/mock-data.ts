import type { Pokemon } from "@/types/pokemon";

export const mockPikachu: Pokemon = {
    id: 25,
    name: "pikachu",
    frenchName: "Pikachu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png",
    },
    types: [
      {
        type: {
          name: "electric",
        },
      },
    ],
    height: 4,
    weight: 60,
    stats: [
      {
        base_stat: 35,
        stat: {
          name: "hp",
        },
      },
    ],
  };
  
export const mockCharizard: Pokemon = {
    id: 6,
    name: "charizard",
    frenchName: "Dracaufeu",
    sprites: {
      front_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
      back_default:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
      front_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
      back_shiny:
        "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
    },
    types: [
      {
        type: {
          name: "fire",
        },
      },
      {
        type: {
          name: "flying",
        },
      },
    ],
    height: 17,
    weight: 905,
    stats: [
      {
        base_stat: 78,
        stat: {
          name: "hp",
        },
      },
    ],
  };


export const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};