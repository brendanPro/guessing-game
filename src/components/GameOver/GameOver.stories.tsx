import '../index.css';
import type { Meta, StoryObj } from '@storybook/react';
import { GameOver } from './GameOver';
import type { Pokemon } from '@/types/pokemon';


const mockPikachu: Pokemon = {
  id: 25,
  name: 'pikachu',
  frenchName: 'Pikachu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png',
  },
  types: [
    {
      type: {
        name: 'electric',
      },
    },
  ],
  height: 4,
  weight: 60,
  stats: [
    {
      base_stat: 35,
      stat: {
        name: 'hp',
      },
    },
  ],
};

const mockCharizard: Pokemon = {
  id: 6,
  name: 'charizard',
  frenchName: 'Dracaufeu',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png',
  },
  types: [
    {
      type: {
        name: 'fire',
      },
    },
    {
      type: {
        name: 'flying',
      },
    },
  ],
  height: 17,
  weight: 905,
  stats: [
    {
      base_stat: 78,
      stat: {
        name: 'hp',
      },
    },
  ],
};

const mockMewtwo: Pokemon = {
  id: 150,
  name: 'mewtwo',
  frenchName: 'Mewtwo',
  sprites: {
    front_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150.png',
    back_default: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/150.png',
    front_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/150.png',
    back_shiny: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/150.png',
  },
  types: [
    {
      type: {
        name: 'psychic',
      },
    },
  ],
  height: 20,
  weight: 1220,
  stats: [
    {
      base_stat: 106,
      stat: {
        name: 'hp',
      },
    },
  ],
};

const meta = {
  title: 'Game/GameOver',
  component: GameOver,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    pokemon: {
      control: 'object',
      description: 'The Pokemon that the player failed to guess',
    },
  },
} satisfies Meta<typeof GameOver>;

export default meta;
type Story = StoryObj<typeof meta>;

export const WithPikachu: Story = {
  args: {
    pokemon: mockPikachu,
  },
};

export const WithCharizard: Story = {
  args: {
    pokemon: mockCharizard,
  },
};

export const WithMewtwo: Story = {
  args: {
    pokemon: mockMewtwo,
  },
};

export const WithLongName: Story = {
  args: {
    pokemon: {
      ...mockCharizard,
      frenchName: 'Tortank-MÃ©ga',
    },
  },
};

export const WithShortName: Story = {
  args: {
    pokemon: {
      ...mockPikachu,
      frenchName: 'Mew',
    },
  },
};

export const NullPokemon: Story = {
  args: {
    pokemon: null,
  },
};

