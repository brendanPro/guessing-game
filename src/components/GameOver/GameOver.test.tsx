import { describe, test, expect, beforeAll } from "bun:test";
import { renderToStaticMarkup } from "react-dom/server";
import { GameOver } from "./GameOver";
import type { Pokemon } from "@/types/pokemon";

// Mock data
const mockPikachu: Pokemon = {
  id: 25,
  name: "pikachu",
  frenchName: "Pikachu",
  sprites: {
    front_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png",
    back_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png",
    front_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/25.png",
    back_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/25.png",
  },
  types: [
    {
      type: {
        name: "electric",
      },
    },
  ],
  height: 4,
  weight: 60,
  stats: [
    {
      base_stat: 35,
      stat: {
        name: "hp",
      },
    },
  ],
};

const mockCharizard: Pokemon = {
  id: 6,
  name: "charizard",
  frenchName: "Dracaufeu",
  sprites: {
    front_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
    back_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
    front_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
    back_shiny:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
  },
  types: [
    {
      type: {
        name: "fire",
      },
    },
    {
      type: {
        name: "flying",
      },
    },
  ],
  height: 17,
  weight: 905,
  stats: [
    {
      base_stat: 78,
      stat: {
        name: "hp",
      },
    },
  ],
};

describe("GameOver Component", () => {
  test("snapshot with Pikachu", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockPikachu} />);
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Charizard (French name)", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockCharizard} />);
    expect(html).toMatchSnapshot();
  });

  test("snapshot with null pokemon", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={null} />);
    expect(html).toMatchSnapshot();
  });

  test("contains game over emoji and text", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockPikachu} />);
    expect(html).toContain("ðŸ˜ž Game Over");
  });

  test("displays pokemon french name", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockPikachu} />);
    expect(html).toContain("Pikachu");
  });

  test("displays correct french name for Charizard", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockCharizard} />);
    expect(html).toContain("Dracaufeu");
  });

  test("has correct CSS classes", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockPikachu} />);
    expect(html).toContain("border-red-500");
    expect(html).toContain("bg-red-50");
  });

  test("contains strong tag for pokemon name", () => {
    const html = renderToStaticMarkup(<GameOver pokemon={mockPikachu} />);
    expect(html).toContain("<strong>");
    expect(html).toContain("</strong>");
  });
});
