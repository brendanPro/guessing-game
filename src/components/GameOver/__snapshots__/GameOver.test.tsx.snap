// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver Component snapshot with Pikachu 1`] = `"<div data-slot="card" class="text-card-foreground rounded-xl border shadow-sm border-red-500 bg-red-50 dark:bg-red-950"><div data-slot="card-content" class="p-4 text-center"><div class="text-red-600 dark:text-red-400"><div class="text-2xl font-bold mb-2">ðŸ˜ž Game Over</div><div>Le Pokemon Ã©tait <strong>Pikachu</strong></div></div></div></div>"`;

exports[`GameOver Component snapshot with Charizard (French name) 1`] = `"<div data-slot="card" class="text-card-foreground rounded-xl border shadow-sm border-red-500 bg-red-50 dark:bg-red-950"><div data-slot="card-content" class="p-4 text-center"><div class="text-red-600 dark:text-red-400"><div class="text-2xl font-bold mb-2">ðŸ˜ž Game Over</div><div>Le Pokemon Ã©tait <strong>Dracaufeu</strong></div></div></div></div>"`;

exports[`GameOver Component snapshot with null pokemon 1`] = `"<div data-slot="card" class="text-card-foreground rounded-xl border shadow-sm border-red-500 bg-red-50 dark:bg-red-950"><div data-slot="card-content" class="p-4 text-center"><div class="text-red-600 dark:text-red-400"><div class="text-2xl font-bold mb-2">ðŸ˜ž Game Over</div><div>Le Pokemon Ã©tait <strong></strong></div></div></div></div>"`;
